imx/signers/stark/factory.go

```diff
@@ -145,3 +282,32 @@ func getStarkPath(layerName, applicationName, ethereumAddress, index string) str
 
 	return fmt.Sprintf("m/2645'/%d'/%d'/%d'/%d'/%s", layer, application, ethAddress1, ethAddress2, index)
 }
+
+// getStarkPublicKeyFromImx gets the account (stark public key) value of the requested user (ethAddress) for Production environment only.
+func getStarkPublicKeyFromImx(ctx context.Context, ethAddress string) (starkPublicKey string, accountNotFound bool, err error) {
+	apiConfig := api.NewConfiguration()
+	apiConfig.Servers[0] = api.ServerConfiguration{
+		URL:         imx.Mainnet.BaseAPIPath,
+		Description: "Prod Environment",
+	}
+	usersAPI := api.NewAPIClient(apiConfig).UsersApi
+
+	// Query existing account value for the given user (ethAddress).
+	response, httpResponse, err := usersAPI.GetUsers(ctx, ethAddress).Execute()
+	defer httpResponse.Body.Close()
+	if err != nil {
+		if httpResponse.StatusCode == http.StatusNotFound {
+			return "", true, nil // This means a new account. So lets use the value from default GrindKey function.
+		}
+		return "", false, imx.NewIMXError(httpResponse, err)
+	}
+	// err is nil from here on
+	if response == nil {
+		return "", false, imx.NewIMXError(httpResponse, errors.New("nil response"))
+	}
+	accounts := response.GetAccounts()
+	if len(accounts) == 0 {
+		return "", false, imx.NewIMXError(httpResponse, errors.New("successful API call, but 0 accounts returned"))
+	}
+	return accounts[0], false, nil
```

Comment:
To cover all invariants we could check for len(accounts) != 1, it shouldn't exist

Reply:
Good point @igx-C,

I would prefer to keep it as is, to keep it consistent with the released `core-sdk` and `sdk-js` versions

Reply:
👍